src/main/scala/is/hail/asm4s/FunctionBuilder.scala

```diff
@@ -143,63 +143,59 @@ class FunctionBuilder[F >: Null](parameterTypeInfo: Array[MaybeGenericTypeInfo[_
     c.emit(l)
   }
 
-  def classAsBytes(): Array[Byte] = {
+  def emit(insn: AbstractInsnNode) {
+    l += insn
+  }
+
+  def classAsBytes(print: Option[PrintWriter] = None): Array[Byte] = {
     mn.instructions.add(start)
     val dupes = l.groupBy(x => x).map(_._2.toArray).filter(_.length > 1).toArray
     assert(dupes.isEmpty, s"some instructions were repeated in the instruction list: ${dupes: Seq[Any]}")
     l.foreach(mn.instructions.add _)
     mn.instructions.add(new InsnNode(returnTypeInfo.returnOp))
     mn.instructions.add(end)
 
-    // The following block of code can help when the output of Verification 2 is
```

Comment:
This is unrelated to the compiler changes, but was necessary to easily debug the compiler.

How it works now:

Attempt to verify correctness of byte code using automatically computed maxes and stack frames. 
 - If that prints no messages and throws no exceptions, then everything worked fine.
 - If that throws no exception, but prints a message, then verification failed. Print the verification failure message and throw an exception.
 - If that throws an exception, then it is likely that auto-computing stack frames failed, so let's try to verify again without the stack frames
  - If that prints no messages, then everything worked fine.
  - If that prints a message, then verification failed. Print the verification failure message and throw an exception.
  - I've never seen the max-only verifier throw an exception.